test/xpu_api/numerics/complex.number/cmplx.over/real.pass.cpp

```diff
@@ -12,6 +12,11 @@
 //   T
 //   real(const T& x);
 
+// Warning	G2904547A	'complex' is deprecated: warning STL4037:
+// The effect of instantiating the template std::complex for any type other than float, double, or long double is unspecified.
+// You can define _SILENCE_NONFLOATING_COMPLEX_DEPRECATION_WARNING to suppress this warning. [-Wdeprecated-declarations]
+#define _SILENCE_NONFLOATING_COMPLEX_DEPRECATION_WARNING
+
```

Comment:
Do we still want to test these types other than float, double, long double using `std::complex<>` if they are unspecified and compilation may fail according to the standard? 

If so, is there some way to check compiler support and enable testing only where it exists rather than just silencing the warning?

Reply:
Fixed